lib/parameters.py

```diff
@@ -181,6 +181,30 @@ def gen_avc_vbr_la_parameters(spec, profiles):
   params = gen_avc_vbr_la_variants(spec, profiles)
   return keys, params
 
+def gen_avc_forced_idr_variants(spec, profiles, tested):
+  for case, params in spec.items():
+    for variant in copy.deepcopy(params.get("forced_idr", [])):
+      uprofile = variant.get("profile", None)
+      cprofiles = [uprofile] if uprofile else profiles
+      rcmode = variant.get("rcmode", None)
+      variant.update(
+        bitrate = None if "cqp" == rcmode else variant["bitrate"],
+        maxrate = None if "cqp" == rcmode else (variant["bitrate"] if "cbr" == rcmode else variant["bitrate"] * 2),
+        qp = None if "cqp" != rcmode else variant["qp"])
+      for profile in cprofiles:
+        testcase = [case, variant["rcmode"], variant["bitrate"],
+          variant["maxrate"], variant["qp"],variant["quality"], profile]
+        #Ensure that there are no duplicate testcase
+        if testcase not in tested:
+          tested.append(testcase)
+          yield testcase
+
+def gen_avc_forced_idr_parameters(spec, profiles):
+  keys = ("case", "rcmode", "bitrate", "maxrate", "qp", "quality", "profile")
+  tested = []
+  params = gen_avc_forced_idr_variants(spec, profiles, tested)
```

Comment:
There is no need to pass `tested` as a parameter... it only needs to be a local inside `gen_avc_forced_idr_variants`.  Nonetheless, `tested` goes away with assert/exception logic as described.